runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -120,7 +129,8 @@ public void write(IToken token, Object data) throws OverwriteException {
         });
 
         // Cache the successful write
-        if (!runtime.isCacheDisabled()) {
+        if ((opts == null || opts != null && opts.isDoCache() )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)